app/containers/IntentService.jsx

```diff
@@ -0,0 +1,102 @@
+/** @jsx h */
+import { h, Component } from 'preact'
+
+import Loading from '../components/Loading'
+
+export default class IntentService extends Component {
+  constructor (props, context) {
+    super(props, context)
+    this.store = context.store
+
+    const {window} = props
```

Comment:
@y-lohse My point too. Bonus part : the intent id can then be directly get from `window.location`, and allow app developer to not do this part (always the same) by themselves.